db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OptimizerImpl.java

```diff
@@ -2404,7 +2426,10 @@ private void ruleBasedCostOptimizable(Optimizable optimizable,
                 bestAp.setLockMode(currentAccessPath.getLockMode());
                 bestAp.setMissingHashKeyOK(currentAccessPath.isMissingHashKeyOK());
                 bestAp.setNumUnusedLeadingIndexFields(currentAccessPath.getNumUnusedLeadingIndexFields());
-
+                bestAp.setUisPredicate(currentAccessPath.getUisPredicate());
+                bestAp.setUisRowIdPredicate(currentAccessPath.getUisRowIdPredicate());
+                bestAp.setUnionOfIndexes(currentAccessPath.getUnionOfIndexes());
+                bestAp.setUisRowIdJoinBackToBaseTableResultSet(currentAccessPath.getUisRowIdJoinBackToBaseTableResultSet());
```

Comment:
It looks like that it makes sense to have a smaller helper function like `setUisFields()`.

Reply:
Yes, that would make it easier to read.  Added setUisFields.